src/main/java/com/team766/orin/GetOrinRawValue.java

```diff
@@ -0,0 +1,132 @@
+package com.team766.orin;
+
+import edu.wpi.first.networktables.DoubleArrayEntry;
+import edu.wpi.first.networktables.DoubleArrayTopic;
+import edu.wpi.first.networktables.IntegerArrayEntry;
+import edu.wpi.first.networktables.IntegerArrayTopic;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+
+/*
+ * This class is used to get raw values from the NetworkTable. We can send values on the Orin to NetworkTables and get them easily here.
+ * One important caveat of the NetworkTables code is that it doesn't throw exceptions when a value is not found.
+ * Instead, it asks for a default value to return. This is a problem because we can't know if the value is not found or if the value is the default value.
+ * Thus, numbers were chosen as default values that aren't really plausible to be sent by the Orin.
+ * For example, it is not likley that the Orin will send Integer.MIN_VALUE as a value. In this case, an exception is thrown.
+ *
+ * @author Max Spier, 9/28/2024
+ */
+public class GetOrinRawValue {
+
+    private static NetworkTableInstance inst = NetworkTableInstance.getDefault();
+    private static NetworkTable table = inst.getTable("/SmartDashboard");
+
+    private static DoubleArrayTopic pose = table.getDoubleArrayTopic("raw_pose");
+    private static double[] arr = new double[] {Double.NEGATIVE_INFINITY};
+    private static DoubleArrayEntry poseValues = pose.getEntry(arr);
+
+    private static IntegerArrayTopic tagId = table.getIntegerArrayTopic("tag_id");
+    private static long[] array1 = new long[] {0l};
+    private static IntegerArrayEntry tagIdValues = tagId.getEntry(array1);
+
+    public static double[] getRawPoseData() throws ValueNotFoundOnTableError {
+        if (poseValues.get().length == 1 && poseValues.get()[0] == Double.NEGATIVE_INFINITY) {
+            throw new ValueNotFoundOnTableError("Pose Data Not Present on Table");
+        }
+
+        return poseValues.get();
+    }
+
+    public static int[] getTagIds() throws ValueNotFoundOnTableError {
+        long[] longValues = tagIdValues.get();
+
+        if (longValues.length == 0
+                || (longValues[0] == Integer.MIN_VALUE && longValues.length == 1)) {
```

Comment:
why not make the default value be an empty array? then you only have one case to check here

Reply:
the networktables table could just be returning an empty array. If that happens we would have no way to check the empty array against the default array.

Reply:
fixed (per conversation in person). TYSM!!